### PR TITLE
KEP-4358: Custom Resource Field Selectors: Promote to Beta

### PR DESCRIPTION
**Pull request by jpbetz**
Update feature gate to reflect that this is in beta starting with 1.31 and is enabled by default.

This feature was accurately documented for alpha in https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/ and https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/, so we don't need to change any other documentation.

### CODE REVIEW

File: content/en/docs/reference/command-line-tools-reference/feature-gates/custom-resource-field-selectors.md
Comment (dipesh-rawat) [inline content/en/docs/reference/command-line-tools-reference/feature-gates/custom-resource-field-selectors.md:14]: Adding `toVersion` to the previous stage would enhance consistency with other feature gate files.

```suggestion
    toVersion: "1.30"
  - stage: beta
    defaultValue: true
    fromVersion: "1.31"  
```

Comment (jpbetz) [reply content/en/docs/reference/command-line-tools-reference/feature-gates/custom-resource-field-selectors.md:15]: Thanks, I had missed that.

**Comment by T-Lakshmi** [comment]
[Preview page](https://deploy-preview-46980--kubernetes-io-main-staging.netlify.app/docs/reference/command-line-tools-reference/feature-gates/#feature-gates-for-alpha-or-beta-features)

/lgtm

**Comment by T-Lakshmi** [comment]
/lgtm cancel

@jpbetz,
It looks documentation update is required in [custom-resource-definitions](https://deploy-preview-46980--kubernetes-io-main-staging.netlify.app/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#crd-selectable-fields) page.

**Comment by T-Lakshmi** [comment]
image

We generally mention only the latest stage of feature gates, please remove the highlight for Alpha stage.
Also please mention this feature gate is default enable since V1.31

**Review by T-Lakshmi — COMMENTED** [review]
/label tide/merge-method-squash

**Comment by sftim** [comment]
/remove-label tide/merge-method-squash

2 commits is OK; 10 is definitely a problem.

**Comment by sftim** [comment]
@jpbetz if you want to squash on your branch and force-push, you can, but it's not required.

See https://github.com/kubernetes/website/pull/46980#issuecomment-2192139838 for a further required update. You won't need to enable the feature gate for v1.31

**Comment by jpbetz** [comment]
> /lgtm cancel
> 
> Joe Betz, It looks documentation update is required in [custom-resource-definitions](https://deploy-preview-46980--kubernetes-io-main-staging.netlify.app/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#crd-selectable-fields) page.

Thanks for catching. I've cleaned up both places.

File: content/en/docs/concepts/extend-kubernetes/api-extension/custom-resources.md
Comment (T-Lakshmi) [inline content/en/docs/concepts/extend-kubernetes/api-extension/custom-resources.md:317]: To clarify more, it's helpful to specify:
1. Which feature gate is responsible for providing this feature.
2. And this feature enabled by default (as its a new update).

WDYT @jpbetz?

```suggestion
declare which other fields in a custom resource may be used in field selectors
with the feature of `CustomResourceFieldSelectors`
[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) (This feature gate is enabled by default since Kubernetes v1.31).
```

File: content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions.md
Comment (T-Lakshmi) [inline content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions.md:1651]:
```suggestion
declare which other fields in a custom resource may be used in field selectors
with the feature of `CustomResourceFieldSelectors`
[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) (This feature gate is enabled by default since Kubernetes v1.31).
```
To clarify more, it's helpful to specify:
1. which feature gate is responsible for providing this feature.
2. And this feature enabled by default (as its a new update).

WDYT @jpbetz?

**Review by T-Lakshmi — CHANGES_REQUESTED** [review]

File: content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions.md
Comment (jpbetz) [reply content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions.md:1649]: Accepted. It would be amazing if we could somehow get this type of explanation uniformly via the `{{< feature-state...`.

Comment (sftim) [reply content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions.md:1649]: Hard to do that with good localization, but PRs are welcome.

**Review by Princesso — COMMENTED** [review]
/lgtm

**Comment by chanieljdan** [comment]
Hi @jpbetz, 

Reaching out from the Docs team, just checking in as we approach Docs Freeze on Tuesday July 30th 18:00 PDT. This documentation appears to still be under review. To meet the Docs Freeze, this PR must have a technical review as well as `lgtm` and `approve` labels applied. Looks like the `lgtm` is in place, we'll just need a technical review as well before sending it for approval. 

Thank you!

Daniel

**Comment by jpbetz** [comment]
All feedback has been applied on this.  I believe it's ready to merge and just needs an approver.

**Pull request merged**
